components/switch/index.tsx

```diff
@@ -66,28 +67,49 @@ export default class Switch extends React.Component<SwitchProps, {}> {
     this.rcSwitch.blur();
   }
 
-  renderSwitch = ({ getPrefixCls }: ConfigConsumerProps) => {
-    const { prefixCls: customizePrefixCls, size, loading, className = '', disabled } = this.props;
+  renderSwitch = ({
+    getPrefixCls,
+    disableAnimation: disableAnimationGlobal,
+  }: ConfigConsumerProps) => {
+    const {
+      prefixCls: customizePrefixCls,
+      size,
+      loading,
+      className = '',
+      disabled,
+      disableAnimation: disableAnimationLocal,
+    } = this.props;
     const prefixCls = getPrefixCls('switch', customizePrefixCls);
+
+    // Give preference to the the prop if it alters the
+    // animation state of the component directly.
+    const disableAnimation =
+      typeof disableAnimationLocal !== 'undefined' ? disableAnimationLocal : disableAnimationGlobal;
```

Comment:
Why is it necessary to use the `typeof`, instead of a simpler check against `undefined`?